sound/soc/sof/topology.c

```diff
@@ -2861,71 +2869,89 @@ static int sof_set_dai_config(struct snd_sof_dev *sdev, u32 size,
 	 * or none of them. Here print a warning message to notify user
 	 */
 	if (!found) {
+		kfree(config);
```

Comment:
I guess this is safe, but it's a bit non-obvious to see this is correct as the alloc has been done two functions up the callstack. Would it be possible to return error here, so the memory would be freed in sof_link_load() ? That would be easier to veriy-to-be-correct by reading the code.

Reply:
I agree, I think we should consolidate the return paths for all errors. Why deal with this one separately?

Reply:
ok, but what is the functionality of this whole thing? This code was there previously and it is not returning error in this case... so is it ok to return error here? the parsed config is "thrown away" with this and previous versions so what is the point here?

Reply:
yeah it didnt make sense before. So lets fix it

Reply:
I changed this now to return an error.